src/code/z_sub_s.c

```diff
@@ -214,17 +213,252 @@ void SubS_UpdateFlags(u16* flags, u16 setBits, u16 unsetBits) {
     *flags = (*flags & ~unsetBits) | setBits;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sub_s/func_8013AF00.s")
+/**
+ * Fills the weightArray to be used with time paths
+ *
+ * @param weightArray an array of values that are used to compute the weightVal and the individual weights
+ * @param numPoints the number of points considered with weights
+ * @param len the length to fill the array, generally the path count + numPoints
+ *
+ * @note `SubS_TimePathing_Update` assumes numPoints to be 3
+ */
+void SubS_TimePathing_FillWeightArray(f32 weightArray[], s32 numPoints, s32 len) {
+    s32 i;
+    f32 val = 0.0f;
+
+    for (i = 0; i < len; i++) {
+        if ((i >= numPoints) && (i < (len - numPoints + 1))) {
+            val += 1.0f;
+        }
+        weightArray[i] = val;
+    }
+}
+
+/**
+ * Computes the weightVal to be used with time paths
+ *
+ * @param weightVal the main weight value used to compute the weights for the points considered
+ * @param elapsedTime how much time has passed
+ * @param waypointTime how much time per each waypoint
+ * @param totalTime how much time the path should take to travel
+ * @param pathCount the path count
+ * @param numPoints the number of points considered when computing the next point to move to
+ * @param weightArray see `SubS_TimePathing_FillWeightArray`
+ *
+ * @return s32 0 for error, 1 if still on the path, and 2 if the end of the path should be reached
+ */
+s32 SubS_TimePathing_ComputeWeightVal(f32* weightVal, s32 elapsedTime, s32 waypointTime, s32 totalTime, s32 pathCount,
+                                      s32 numPoints, f32 weightArray[]) {
```

Comment:
That makes sense. I think progress seems best since it represents how far along, the progress, along the given path.
This also seems to make sense for the `WeightPathing` systems, though that seems more like progress towards a given waypoint rather than the path as a whole.